pkg/targetgroupbinding/resource_manager.go

```diff
@@ -461,3 +475,26 @@ func isELBV2TargetGroupNotFoundError(err error) bool {
 	}
 	return false
 }
+
+func isELBV2TargetInELBVPC(podIP string, vpc *ec2sdk.Vpc) bool {
+	// Check if the pod IP is in the primary VPC CIDR.
+	if isIPinCIDR(podIP, *vpc.CidrBlock) {
+		return true
+	}
+
+	// Check if the pod IP is from a secondary CIDR block.
+	for _, v := range vpc.CidrBlockAssociationSet {
+		if isIPinCIDR(podIP, *v.CidrBlock) {
+			return true
+		}
+	}
+
+	// Cannot find pod IP in a VPC CIDR.
+	return false
+}
+
+func isIPinCIDR(ipAddr, cidrBlock string) bool {
+	_, cidr, _ := net.ParseCIDR(cidrBlock)
```

Comment:
let's not ignore the err in both net.ParseCIDR and net.ParseIP. (though it's unlikely to happen in current code)

we can only return true if err in both case is nil.